velox/exec/prefixsort/PrefixSortEncoder.h

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <cstdint>
+#include <functional>
+#include <memory>
+
+#include "velox/common/base/BitUtil.h"
+#include "velox/common/base/Exceptions.h"
+#include "velox/common/base/SimdUtil.h"
+
+namespace facebook::velox::exec::prefixsort {
+
+/// Provides encode/decode methods for PrefixSort.
+class PrefixSortEncoder {
+ private:
+  FOLLY_ALWAYS_INLINE static uint8_t flipSignBit(uint8_t byte) {
+    return byte ^ 128;
+  }
+
+ public:
+  template <typename T>
+  static FOLLY_ALWAYS_INLINE void encode(T value, char* row) {
```

Comment:
Please, add comments to explain what types are supported and how they are encoded. I.e. clarify that only int64_t is supported and encoding is compatible with sorting ascending with no nulls.

Reply:
@mbasmanova Tanks a lot ~